test/unit/org/apache/cassandra/service/reads/range/ReplicaPlanMergerTest.java

```diff
@@ -0,0 +1,486 @@
+                       range(endOf("10"), endOf("20")),
+                       range(endOf("20"), endOf("30")),
+                       range(endOf("30"), endOf("40")),
+                       range(endOf("40"), rp("")));
+            testRanges(cl,
+                       range(rp(""), rp("25")),
+                       range(rp(""), endOf("10")),
+                       range(endOf("10"), endOf("20")),
+                       range(endOf("20"), rp("25")));
+            testRanges(cl,
+                       range(rp(""), rp("40")),
+                       range(rp(""), endOf("10")),
+                       range(endOf("10"), endOf("20")),
+                       range(endOf("20"), endOf("30")),
+                       range(endOf("30"), rp("40")));
+            testRanges(cl,
+                       range(rp(""), rp("50")),
+                       range(rp(""), endOf("10")),
+                       range(endOf("10"), endOf("20")),
+                       range(endOf("20"), endOf("30")),
+                       range(endOf("30"), endOf("40")),
+                       range(endOf("40"), rp("50")));
+            testRanges(cl,
+                       range(rp("20"), rp("30")),
+                       range(rp("20"), endOf("20")),
+                       range(endOf("20"), rp("30")));
+            testRanges(cl,
+                       range(rp("25"), rp("")),
+                       range(rp("25"), endOf("30")),
+                       range(endOf("30"), endOf("40")),
+                       range(endOf("40"), rp("")));
+            testRanges(cl,
+                       range(rp("50"), rp("")));
+            testRanges(cl,
+                       range(rp("30"), rp("10")),
+                       range(rp("30"), endOf("30")),
+                       range(endOf("30"), endOf("40")),
+                       range(endOf("40"), rp("")),
+                       range(rp(""), rp("10"))); // wrapped
+            testRanges(cl,
+                       range(rp("25"), rp("15")),
+                       range(rp("25"), endOf("30")),
+                       range(endOf("30"), endOf("40")),
+                       range(endOf("40"), rp("")),
+                       range(rp(""), endOf("10")),
+                       range(endOf("10"), rp("15"))); // wrapped
+        }
+    }
+
+    /**
+     * Tests range merging with a multinode cluster and a read consistency level that allows to merge ranges,
+     * particularly when the token ranges don't overlap between replicas.
+     */
+    @Test
+    public void testMultiNodeWithMergingConsistencyLevelContinuous() throws UnknownHostException
+    {
+        new TokenUpdater().withTokens(InetAddressAndPort.getByName("127.0.0.1"), "10", "20", "30")
+                          .withTokens(InetAddressAndPort.getByName("127.0.0.2"), "40", "50", "60")
+                          .withTokens(InetAddressAndPort.getByName("127.0.0.3"), "70", "80", "90")
+                          .update();
```

Comment:
@adelapena These tests throw a bunch of errors at the logs for the missing endpoints in `Gossiper`. We might have to either initialize them (via `Gossiper#initializeNodeUnsafe()` a la `LeaveAndBootstrapTest`) or find a way to mock out the `ReplicaLayout` liveness check.

```
ERROR [main] 2020-11-17 16:00:31,012 FailureDetector.java:281 - Unknown endpoint: /127.0.0.3:7012
java.lang.IllegalArgumentException: 
	at org.apache.cassandra.gms.FailureDetector.isAlive(FailureDetector.java:281)
	at org.apache.cassandra.gms.FailureDetector.lambda$static$0(FailureDetector.java:76)
	at org.apache.cassandra.locator.AbstractReplicaCollection.filter(AbstractReplicaCollection.java:427)
	at org.apache.cassandra.locator.AbstractReplicaCollection.filter(AbstractReplicaCollection.java:412)
	at org.apache.cassandra.locator.ReplicaLayout.forRangeReadLiveSorted(ReplicaLayout.java:335)
	at org.apache.cassandra.locator.ReplicaPlans.forRangeRead(ReplicaPlans.java:606)
	at org.apache.cassandra.service.reads.range.ReplicaPlanIterator.computeNext(ReplicaPlanIterator.java:76)
	at org.apache.cassandra.service.reads.range.ReplicaPlanIterator.computeNext(ReplicaPlanIterator.java:42)
	at org.apache.cassandra.utils.AbstractIterator.hasNext(AbstractIterator.java:47)
	at com.google.common.collect.Iterators$PeekingImpl.hasNext(Iterators.java:1126)
	at org.apache.cassandra.service.reads.range.ReplicaPlanMerger.computeNext(ReplicaPlanMerger.java:56)
	at org.apache.cassandra.service.reads.range.ReplicaPlanMerger.computeNext(ReplicaPlanMerger.java:32)
```

Reply:
I think this is actually a bug in a tests as well. Without `.2` and `.3` visible to the `Gossiper`, the `ReplicaPlanIterator` produces replica plans that have no candidate endpoints, and that means we merge much less than we should. One other way we might be able to see this is by having the test extend `CQLTester` and calling `requireNetwork()` after `prepareServer()`.

Reply:
You are totally right, I have added those calls to `Gossiper#initializeNodeUnsafe` and changed the range tests, that weren't calculating the tokens properly.

Reply:
@adelapena These changes look great. Thanks!